nengo/cache.py

```diff
@@ -284,16 +285,21 @@ def _check_legacy_file(self):
         legacy_file = os.path.join(self.cache_dir, self._LEGACY)
         if os.path.exists(legacy_file):
             with open(legacy_file, 'r') as lf:
-                version = int(lf.read().strip())
+                text = lf.read()
+            try:
+                lv, pp = tuple(int(x.strip()) for x in text.split('.'))
+            except ValueError:
+                # Will be raised with old legacy.txt format
+                lv = pp = -1
         else:
-            version = -1
-        return version == self._LEGACY_VERSION
+            lv = pp = -1
+        return lv == self._LEGACY_VERSION and pp == self._PICKLE_PROTOCOL
```

Comment:
I'd just wipe the cache whenever things change.  Wiping the cache is not that bad, just (maybe) makes things slightly slower on the next build.  Seems worth it to just keep things clean and not  have to worry about much more disruptive errors.